Modules/Numerics/Optimizersv4/test/itkRegularStepGradientDescentOptimizerv4Test.cxx

```diff
@@ -384,6 +384,50 @@ itkRegularStepGradientDescentOptimizerv4Test(int, char *[])
                                                                                 currentLearningRateRelaxation);
   }
 
+  // Verify that the optimizer stays at the initial position if
+  // number of iterations is set to one.
+  std::cout << "\nCheck the optimizer position (parameters) are equal to the initial "
+               "value when number of iterations is set to one:"
+            << std::endl;
+  {
+    RSGv4TestMetric::Pointer metric = RSGv4TestMetric::New();
+    itkOptimizer->SetMetric(metric);
+    using ParametersType = RSGv4TestMetric::ParametersType;
+    const unsigned int spaceDimension = metric->GetNumberOfParameters();
+    ParametersType     initialPosition(spaceDimension);
+    initialPosition[0] = 100;
+    initialPosition[1] = -100;
+    metric->SetParameters(initialPosition);
+
+    itkOptimizer->SetNumberOfIterations(1);
```

Comment:
The existing behavior is:

Zero iterations: No function evaluations performed
One iteration: One function evaluation, update position (Parameters)

The new behavior is:

Zero iterations: No function evaluations performed
One iteration: One function evaluation, do not update position (Parameters)

So it depends upon what is the definition of an iteration. It would be helpful to have documentation for what is supposed to happen.